tests/installation/ipxe_install.pm

```diff
@@ -131,26 +131,43 @@ END_BOOTSCRIPT
     diag "$response->{status} $response->{reason}\n";
 }
 
+
 sub run {
     my $self = shift;
 
     poweroff_host;
 
     #virtualization tests use a static ipxe configuration file
-    set_bootscript unless get_var('VIRT_AUTOTEST');
+    set_bootscript unless get_var('VIRT_AUTOTEST') && is_tumbleweed;
 
     set_pxe_boot;
 
     poweron_host;
 
+    select_console 'sol', await_console => 0;
+
+    #use ipxe bootloader in O3
+    #it is static menu and choose the TW entry to start installation
+    if (get_var('VIRT_AUTOTEST') && is_tumbleweed) {
+        ipmi_backend_utils::ipmitool 'chassis power reset' if !check_screen([qw(o3-ipxe-menu ipxe-boot-failure)], 300);
+        assert_screen('o3-ipxe-menu', 300) if match_has_tag('ipxe-boot-failure');
+        send_key 't';
+        assert_screen([qw(load-linux-kernel load-initrd)], 240);
+        #loading initrd spend much time(fg. 10-15 minutes to Beijing SUT)
+        assert_screen([qw(start-tw-install)], 360);
+        assert_screen([qw(network-config-created loading-installation-system)], 60);
```

Comment:
Sequence of check_screen and assert_screen without any interaction. I doubt it can function very well.

Reply:
It takes longer to download kernel files from downloads.opensuse.org than SUSE internal network. These asserts are intend to make sure the process is ongoing, and detect failure ASAP.